### DISCUSSION TITLE
SFTP Client Access to WebAdmin Folders?

### DISCUSSION THREAD

**Discussion by jarrelldunson** [drakkan/sftpgo#1140] [General]
Hey, when I login SFTPGo (via elestio) through an SFTP web client (WinSCP), I can only see the root directory.   How can I access the ‘WebAdmin’ Folders I created (e.g.   Local: /foldername) .. in the elestio.app/web/admin/folders list ?
Thanks, Jarrell 
Jarrell_dunson@pfm.org

**Comment by drakkan** [comment]
Hello,

SFTPGo enforces chroot isolation for its users. Take a look at the configured home dir using the SFTPGo web admin UI, you can only access files in this directory. 

To fix the reported issue you can change the user home dir or map paths outside the home dir as [virtual folders](https://github.com/drakkan/sftpgo/blob/main/docs/virtual-folders.md). 

Note that SFTPGo runs, by default, as a non-root user (uid 1000 on Docker), if you change your home directory or map a virtual folder you need to ensure that the user running SFTPGo can read/write there.

**Reply by jarrelldunson** [reply]
Thanks so much for responding - but am unclear on much of this
- I don't see home directory in WebAdmin. In WebAdmin, I have this side menu: Users, Folders, Groups, Event Manager, Connections, Admins, Maintenance, Status (none of these list a configuration and/or 'home directory' for root).
- In Folders, there are some created:  Local:/, Local:/[Folder Name], etc.  
- I was able to login via WinSCP and even create a directory under the root directory (but this new directory doesn't display under the SFTPGo UI (not even Folders) 
- I don't see how to map virtual folders withing SFTPGo 
Can you help explain? and/or link some resources ?  thanks, Jarrell

**Reply by drakkan** [reply]
> Thanks so much for responding - but am unclear on much of this
> 
> * I don't see home directory in WebAdmin. In WebAdmin, I have this side menu: Users, Folders, Groups, Event Manager, Connections, Admins, Maintenance, Status (none of these list a configuration and/or 'home directory' for root).

click on Users and then select a user, you can see the home dir there

![Schermata del 2023-01-11 16-16-21](https://user-images.githubusercontent.com/553263/211843527-c1153e3b-73d3-40e8-a69d-616f351a9337.png)

in you have no users, then you are not connecting to SFTPGo. SFTPGo runs by default on port 2022 and it uses virtual users (no existing system users). Take a look at the [getting started guide](https://github.com/drakkan/sftpgo/blob/main/docs/howto/getting-started.md) to understand how SFTPGo works. 

There is an excellent external guide here

https://ostechnix.com/setup-sftp-server-with-sftpgo-in-linux/

> * In Folders, there are some created:  Local:/, Local:/[Folder Name], etc.
> * I was able to login via WinSCP and even create a directory under the root directory (but this new directory doesn't display under the SFTPGo UI (not even Folders)
> * I don't see how to map virtual folders withing SFTPGo

VIrtual folders are defined in SFTPGo in the "Folders" section. You can map folders to users from the edit user page (the user running SFTPGo must be able to read/write the defined folders)

>   Can you help explain? and/or link some resources ?  thanks, Jarrell
